twitchio/http.py

```diff
@@ -90,23 +90,19 @@ def __init__(self, loop, **attrs):
             return data
 
     @update_bucket
-    async def _get_streams_by_id(self, channels: Sequence[int]):
-        ids = set()
+    async def _get_streams(self, channels: Sequence[Union[str, int]]):
+        args = set()
 
-        # Check if we are only getting ID's not names.
+        # Check if we are only getting ID's or names.
         for chan in channels:
-            try:
-                chan = int(chan)
-            except (TypeError, ValueError):
-                pass
-            else:
-                ids.add(chan)
+            if isinstance(chan, str) or isinstance(chan, int):
```

Comment:
Instead of checking the object type here once or twice depending on the given type, then again in line 104 you could have two separate sets for user_ids and user_logins, then add them to the query string below.

Reply:
It's a good idea to allow passing of argument lists so working with a single stream using the new method is easier and allows removing the "duplicate" methods we have for getting single streams right now later on.

Reply:
Alright I make the change, without removing the single getter if you want to do it later